src/SqlHydra.Query/QueryFunctions.fs

```diff
@@ -9,6 +9,10 @@ module Table =
         let tables = Map [Root, { Name = ent.Name; Schema = ent.DeclaringType.Name}]
         QuerySource<'T>(tables)
 
+    type CorrelatedTable<'T> = CorrelatedTable of QuerySource<'T>
+
+    let correlatedTable<'T> = CorrelatedTable table<'T>
```

Comment:
We could also just make the `correlate` command just work with a `QuerySource<'T>` and the `table` function instead of this.